regions/io/core.py

```diff
@@ -221,3 +371,120 @@ def to_region(self):
+        if self.format_type not in ['CRTF', 'DS9']:
+            raise ValueError("{0} is not available as io".format(self.format_type))
+        if self.region_type not in regions_attributes.keys():
+            raise ValueError("{0} is not a valid region type in this package".format(self.region_type, self.format_type))
+        if self.coordsys not in valid_coordsys[self.format_type]:
+            raise ValueError("{0} is not a valid coordinate reference frame in astropy".
+                             format(self.coordsys, self.format_type))
+
+
+def to_shape_list(region_list, format_type='DS9', coordinate_system='fk5'):
+
+    shape_list = ShapeList()
+
+    for region in region_list:
+
+        coord = []
+        reg_type = str(type(region)).split(".")[2]
+
+        for val in regions_attributes[reg_type]:
+            coord.append(getattr(region, val))
+
+        if reg_type == 'polygon':
+            coord = [x for x in region.vertices]
+
+        coordsys = 'image'
+        if isinstance(region, SkyRegion):
+            coordsys = coord[0].name
+
+        coordsys = coordinate_system or coordsys
+        frame = coordinates.frame_transform_graph.lookup_name(coordsys)
+
+        new_coord = []
+        for val in coord:
+            if isinstance(val, Angle) or isinstance(val, u.Quantity) or isinstance(val, float):
+                new_coord.append(val)
+            elif isinstance(val, PixCoord):
+                new_coord.append(u.Quantity(val.x, u.dimensionless_unscaled))
+                new_coord.append(u.Quantity(val.y, u.dimensionless_unscaled))
+            else:
+                new_coord.append(Angle(val.transform_to(frame).spherical.lon))
+                new_coord.append(Angle(val.transform_to(frame).spherical.lat))
+
+        if format_type == 'DS9':
+            meta = to_ds9_meta(region.meta, region.visual)
+        else:
+            meta = to_crtf_meta(region.meta, region.visual)
+
+        shape_list.append(Shape(format_type, coordsys, reg_type, new_coord, meta, False,
+                                region.meta.get('include', False)))
+
+    return shape_list
+
+
+def to_ds9_meta(region_meta, region_visual):
+
+    # meta keys allowed in DS9
+    valid_keys = ['label', 'symbol', 'include', 'tag', 'line', 'comment',
+                  'name', 'select', 'highlite', 'fixed',
+                  'edit', 'move', 'rotate', 'delete', 'source', 'background']
+
+    # visual keys allowed in DS9
+    valid_keys += ['color', 'dash', 'linewidth', 'font', 'dashlist', 'fill']
+
+    # mapped to actual names in DS9
+    key_mappings = {'symbol': 'point', 'label': 'text', 'linewidth': 'width'}
+
+    meta = dict()
+    for key in region_meta:
+        if key in valid_keys:
+            meta[key_mappings.get(key, key)] = region_meta[key]
+    for key in region_visual:
+        if key in valid_keys:
+            meta[key_mappings.get(key, key)] = region_visual[key]
+
+    return meta
+
+
+def to_crtf_meta(region_meta, region_visual):
```

Comment:
the code in here is nearly identical to that in `to_ds9_meta`.  Consider abstracting out the code and make `valid_keys` and `key_mappings` parameters to a generic `to_meta` function, then have `to_ds9_meta` and `to_crtf_meta` wrap that `to_meta_function`

Reply:
ok! will try.